packages/ionic/src/commands/package/package.ts

```diff
@@ -0,0 +1,344 @@
+import { sleep } from '@ionic/cli-framework/utils/process';
+import chalk from 'chalk';
+import * as Debug from 'debug';
+import * as fs from 'fs';
+import * as https from 'https';
+
+import { CommandMetadata } from '../../definitions';
+import { isSuperAgentError } from '../../guards';
+import { Command } from '../../lib/command';
+import { FatalException } from '../../lib/errors';
+import { fileUtils } from '../../lib/utils/file';
+
+const debug = Debug('ionic:commands:package:build');
+const PLATFORMS = ['android', 'ios'];
+const ANDROID_BUILD_TYPES = ['debug', 'release'];
+const IOS_BUILD_TYPES = ['development', 'ad-hoc', 'app-store', 'enterprise'];
+const BUILD_TYPES = ANDROID_BUILD_TYPES.concat(IOS_BUILD_TYPES);
+
+interface PackageBuild {
+  job_id: number;
+  id: string;
+  caller_id: number;
+  platform: string;
+  build_type: string;
+  created: string;
+  finished: string;
+  state: string;
+  commit: any;
+  stack: any;
+  profile_tag: string;
+  automation_id: number;
+  environment_id: number;
+  native_config_id: number;
+  automation_name: string;
+  environment_name: string;
+  native_config_name: string;
+  job: any;
+}
+
+interface DownloadUrl {
+  url: string | null;
+}
+
+export class BuildCommand extends Command {
+  async getMetadata(): Promise<CommandMetadata> {
+    return {
+      name: 'build',
+      type: 'project',
+      summary: `Creates a package build on Appflow using name parameters, then tails the build log from Appflow and
+finally downloads the created app package file in the current directory if the build is successful.
+
+The basic commands can be customized a combination of Options and Advanced Options.
+
+Apart from the --commit option, all the others options can be specified using the verbose name you selected upon setup in the Appflow Dashboard.
+
+The --security-profile option is mandatory for any iOS build while is not required from an android debug build.
+
+Other notes:
+--environment allows to specify the name of an environment to customize the build
+--native-config allows to specify the name of a native config set of parameters to override the default specified in the app
+--target-platform allows to override the preferred platform with another one: this is currently useful only for building older iOS apps instead of the preferred iOS 10 used by default
+--build-file-name allows to specify a custon name for the build package file that will be downloaded; it can only be a file name and not a path
+`,
+      exampleCommands: [
+        'android debug',
+        'ios development --security-profile="iOS Security Profile Name"',
+        'android debug --environment="My Custom Environment Name"',
+        'android debug --native-config"My Custom Native Config Name"',
+        'android debug --commit=2345cd3305a1cf94de34e93b73a932f25baac77c',
+        'ios development --security-profile="iOS Security Profile Name" --target-platform="iOS - Xcode 9"',
```

Comment:
What is `iOS - Xcode 9`? What are the valid values for `--target-platform`? Is this something setup in Appflow?

Reply:
yes, this is the name (not the id) of the platform on Appflow

![screen shot 2019-01-03 at 9 55 13 pm](https://user-images.githubusercontent.com/1005204/50671834-46e45b00-0fa2-11e9-8655-f3a6cd735e3b.png)

Reply:
Thanks for explaining!